crates/miden-tx/src/executor/mod.rs

```diff
@@ -325,138 +216,78 @@ where
         Ok(*stack_outputs)
     }
 
-    // CHECK CONSUMABILITY
-    // ============================================================================================
+    // HELPER METHODS
+    // --------------------------------------------------------------------------------------------
 
-    /// Validates input notes, transaction inputs, and account inputs before executing the
-    /// transaction with specified notes. Keeps track and returns both successfully consumed notes
-    /// as well as notes that failed to be consumed.
+    /// Prepares the data needed for transaction execution.
     ///
-    /// The `source_manager` is used to map potential errors back to their source code. To get the
-    /// most value out of it, use the source manager from the
-    /// [`Assembler`](miden_objects::assembly::Assembler) that assembled the Miden Assembly code
-    /// that should be debugged, e.g. account components, note scripts or transaction scripts. If
-    /// no error-to-source mapping is desired, a default source manager can be passed, e.g.
-    /// [`DefaultSourceManager::default`](miden_objects::assembly::DefaultSourceManager::default).
-    ///
-    /// # Errors:
-    /// Returns an error if:
-    /// - If required data can not be fetched from the [`DataStore`].
-    /// - If the transaction host can not be created from the provided values.
-    /// - If the execution of the provided program fails on the stage other than note execution.
-    pub(crate) async fn try_execute_notes(
+    /// Preparation includes loading transaction inputs from the data store, validating them, and
+    /// instantiating a transaction host.
+    async fn prepare_transaction(
         &self,
         account_id: AccountId,
         block_ref: BlockNumber,
         notes: InputNotes<InputNote>,
-        tx_args: TransactionArgs,
-    ) -> Result<NoteConsumptionInfo, TransactionExecutorError> {
-        if notes.is_empty() {
-            return Ok(NoteConsumptionInfo::default());
-        }
-        // Validate input notes.
+        tx_args: &TransactionArgs,
+        init_advice_inputs: Option<AdviceInputs>,
+    ) -> Result<
+        (
+            TransactionExecutorHost<'store, 'auth, STORE, AUTH>,
+            TransactionInputs,
+            StackInputs,
+            AdviceInputs,
+        ),
```

Comment:
This looks pretty ugly, but I think it is OK for now. As mentioned in the last comment, we'll probably need to refactor this and so that may be a good opportunity to normalize things.